pkg/providers/openstack/openstack_auth.go

```diff
@@ -0,0 +1,115 @@
+package openstack
+
+import (
+	"fmt"
+	"github.com/emc-advanced-dev/unik/pkg/config"
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack"
+	"os"
+)
+
+type openstackHandle struct {
+	AuthClient *gophercloud.ProviderClient
+	Region     string
+}
+
+// MergeConfWithEnv overrides conf with environment values (inplace).
+func MergeConfWithEnv(conf *config.Openstack) {
+	if v := os.Getenv("OS_AUTH_URL"); v != "" {
+		conf.AuthUrl = v
+	}
+	if v := os.Getenv("OS_USER_ID"); v != "" {
+		conf.UserId = v
+	}
+	if v := os.Getenv("OS_USERNAME"); v != "" {
+		conf.UserName = v
+	}
+	if v := os.Getenv("OS_PASSWORD"); v != "" {
+		conf.Password = v
+	}
+	if v := os.Getenv("OS_TENANT_ID"); v != "" {
+		conf.TenantId = v
+	}
+	if v := os.Getenv("OS_TENANT_NAME"); v != "" {
+		conf.TenantName = v
+	}
+	if v := os.Getenv("OS_DOMAIN_ID"); v != "" {
+		conf.DomainId = v
+	}
+	if v := os.Getenv("OS_DOMAIN_NAME"); v != "" {
+		conf.DomainName = v
+	}
+	if v := os.Getenv("OS_REGION_ID"); v != "" {
+		conf.RegionId = v
+	}
+	if v := os.Getenv("OS_REGION_NAME"); v != "" {
+		conf.RegionName = v
+	}
+}
+
+// obtainCredentials validates credentials then builds gophercloud auth options.
+func obtainCredentials(conf *config.Openstack) (*gophercloud.AuthOptions, error) {
```

Comment:
Ah, no - it doesn't actually authenticate, just validates and builds the opts. Ignore.